### ISSUE TITLE
Cloning a tls cluster

### ISSUE THREAD

**Issue by aurelien43** [CrunchyData/postgres-operator#1677]
Hello, 
when I try to clone a cluster with tls only, the pgo clone commande failed. 
An error appear in the logs of the cluster : `could not load server certificate file`

**To Reproduce**
 
Steps to reproduce the Bug :

1. Create a tls cluster 
2. Clone the cluster 
3. The cloning cluster can't start. 

**Please tell us about your environment:**
* Operating System: centos7
* Where is this running ( Local, Cloud Provider): VM
* Storage being used (NFS, Hostpath, Gluster, etc): NFS
* Container Image Tag: centos7-12.3-4.3.2
* PostgreSQL Version: 12.3
* Platform (Docker, Kubernetes, OpenShift): Kubernetes
* Platform Version: 1.16

Creation of the tls-cluster : 
![tls-create](https://user-images.githubusercontent.com/25898200/86126833-41a49580-badf-11ea-8a38-a35ab31f56e1.PNG)

PGO Clone: 
![clone_create](https://user-images.githubusercontent.com/25898200/86126867-53863880-badf-11ea-9ae5-206657ce050c.PNG)

Error present in pods logs : 
![clone error](https://user-images.githubusercontent.com/25898200/86126906-639e1800-badf-11ea-8509-e5cd1ccf2748.PNG)

**Comment by jkatz** [comment]
Thanks for reporting @aurelien43, I was able to reproduce this issue.

It does raise a good question, i.e. what should the behavior be when a PostgreSQL cluster with `--tls-only` is cloned. Should we create a new TLS secret with the server key/pair of the old server? What if that keypair has a `cn` that is different than the one in the cloned cluster? It may be tough to make that assumption.

The safe route would be to not have TLS required (or enabled?) in the newly cloned cluster. Some of this pain should be offset by the new restore from backup feature (#1679), but this bug should be addressed in 4.3 too.

**Comment by jkatz** [comment]
@aurelien43 OK, I have thought this one through.

I was not particularly happy with #1689 as it leads to copying around Secrets, and we can't necessarily make any assumptions about the contents of the secrets and what they would mean in the new clusters. As such, I don't think `pgo clone` can properly support this.

BUT!

PostgreSQL Operator 4.4 introduces the `--restore-from` flag on `pgo create`, which allows one to create a PostgreSQL cluster from an pgBackRest repo, whether it's tied to an active cluster (e.g. the equivalent of `pgo clone`) or if it's from a pgBackRest repository of an inactive cluster (e.g. the remnants from `pgo delete cluster --keep-backups`). As such, you can perform the equivalent action from your `--tls-only` cluster as such:

```shell
pgo create cluster mycluster-tls-clone --restore-from mycluster-tls
```

And if you wanted your cloned cluster to also be TLS only:

```shell
pgo create cluster mycluster-tls-clone --restore-from mycluster-tls \
  --tls-only \
  --server-ca-secret=postgresql-ca \
  --server-tls-secret=hippo-tls-keypair
```

PostgreSQL Operator 4.4 will be available in the coming days. As such, I will mark this as closed.
